api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -264,4 +295,498 @@ public void saveConcept_shouldForceSetFlagIfSetMembersExist() throws Exception {
 		//Then
 		assertTrue(concept.getSet());
 	}
+	
+	/**
+	 * @see ConceptServiceImpl#purgeConcept(Concept)
+	 * @Verifies should purge the concept if not being used by an obs
+	 */
+	@Test
+	public void purgeConcept_shouldPurgeTheConceptIfNotBeingUsedByAnObs() throws Exception {
+		int conceptId = 88;
+		conceptService.purgeConcept(conceptService.getConcept(conceptId));
+		assertNull(conceptService.getConcept(conceptId));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#retireConcept(Concept,String)
+	 * @Verifies should fail if no reason is given
+	 */
+	@Test
+	public void retireConcept_shouldFailIfNoReasonIsGiven() throws Exception {
+		int conceptId = 3;
+		Concept concept = conceptService.getConcept(conceptId);
+		expectedException.expect(IllegalArgumentException.class);
+		conceptService.retireConcept(concept, "");
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#retireConcept(Concept,String)
+	 * @Verifies should retire the given concept
+	 */
+	@Test
+	public void retireConcept_shouldRetireTheGivenConcept() throws Exception {
+		int drugId = 3;
+		String retireReason = "dummy reason";
+		Concept concept = conceptService.getConcept(drugId);
+		assertFalse(concept.isRetired());
+		conceptService.retireConcept(concept, retireReason);
+		assertTrue(concept.isRetired());
+		assertEquals(retireReason, concept.getRetireReason());
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#purgeDrug(Drug)
+	 * @Verifies should purge the given Drug
+	 */
+	@Test
+	public void purgeDrug_shouldPurgeTheGivenDrug() throws Exception {
+		int drugId = 2;
+		conceptService.purgeDrug(conceptService.getDrug(drugId));
+		assertNull(conceptService.getDrug(drugId));
+	}
+	
+	/**
+	 * @see ConceptServiceImpl#getAllDrugs()
+	 * @Verifies should return a list of all drugs
+	 */
+	@Test
+	public void getAllDrugs_shouldReturnAListOfAllDrugs() throws Exception {
+		int resultWhenTrue = 4;
+		int resultWhenFalse = 2;
+		List<Drug> allDrugs = conceptService.getAllDrugs();
+		assertEquals(resultWhenTrue, allDrugs.size());
+		allDrugs = conceptService.getAllDrugs(false);
```

Comment:
@mavrk one test should only be for one method and one combination of parameters.
and the goal is that every public method in the service layer is tested.

so you should write tests for the methods

* getAllDrugs()
* getAllDrugs(includeRetired)

and for the latter you need one test passing true and one passing false.
another case is also what if there are no drugs will it return an empty list?

I know this is tedious but thats because when this was written [TDD|https://en.wikipedia.org/wiki/Test-driven_development] which would start with tests wasnt used.